lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1105,10 +1113,17 @@ void TNonblockingServer::listenSocket(THRIFT_SOCKET s) {
   serverSocket_ = s;
 
   if (!port_) {
-    sockaddr_in addr;
-    socklen_t size = sizeof(addr);
-    if (!getsockname(serverSocket_, reinterpret_cast<sockaddr*>(&addr), &size)) {
-      listenPort_ = ntohs(addr.sin_port);
+    #ifdef _WIN32
+      struct sockaddr_storage addr;
+      struct sockaddr_in *pAddr = (sockaddr_in *) &addr;
+      socklen_t size = sizeof(addr);
+    #else
+      struct sockaddr_in addr;
+      struct sockaddr_in *pAddr = &addr;
+      socklen_t size = sizeof(addr);
+    #endif
```

Comment:
Well, I do not have an answer, but it seems an official Windows API, cause all "stackoverflow" posts I found are not related to MinGW usage. Even if you search through different blogs & forums, you'll find the same answer, and MinGW is not mentioned.

Moreover, sample code reported in MSDN documentation comments uses getsockname in the same way:

https://msdn.microsoft.com/en-us/library/windows/desktop/ms738543(v=vs.85).aspx